Домашняя работа №-10

NickFilimonov opened the pull request:
Домашняя работа №-10
- Добавлен data class Track с информацией о муз. треке
- Создана mutablelist объектов Track в методе onCreate() SearchActivity, вызван adapter
- Сверстан xml файл track_item и передан в объект RecyclerView файла activity_search.xml
- Созданы реализации классов ViewHolder и Adapter для RecyclerView файла activity_search.xml

NickFilimonov committed:
pr. Playlist Maker commit 12
Домашняя работа №-10
- Добавлен data class Track с информацией о муз. треке
- Создана mutablelist объектов Track в методе onCreate() SearchActivity, вызван adapter
- Сверстан xml файл track_item и передан в объект RecyclerView файла activity_search.xml
- Созданы реализации классов ViewHolder и Adapter для RecyclerView файла activity_search.xml

kirillNay commented on app/src/main/java/com/practicum/playlistmaker/TrackAdapter.kt:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    override fun getItemCount(): Int {
+        return tracks.size
```
⏫ Можно воспользоваться более "котлиновским" вариантом
```
override fun getItemCount(): Int = tracks.size
```

kirillNay commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,31 @@
+        bandName.text = item.artistName
+        trackTime.text = item.trackTime
+        Glide.with(itemView)
+        .load(item.artworkUrl100)
```
⏫ Стоит обратить внимание на отступы, кажется они съехали. В Android Studio есть удобный инструмент автоматического форматирования кода. Для этого достаточно нажать на горячие клавиши **Ctrl+Alt+L** (**Command+Option+L**)

kirillNay commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,31 @@
+        .placeholder(R.drawable.placeholder)
+        .centerCrop()
+        .transform(RoundedCorners(2))
+        .into(albumsCover)
```
⏫ В `RoundedCorners` передается радиус в пикселях. 
Чтобы нам получить значение в пикселях определенное в дименах, нужно в ресурсах - `dimens` определить это значение и обратиться к нему через 
```
itemView.context.resources.getDimensionPixelSize(R.dimen.название поля ресурсов..)
```

kirillNay commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
👍 Отличная идея использовать `ConstraintLayout`! Вообще, `ConstraintLayout` мы будем рассматривать в одном из следующих уроков. Но если интересно, вот очень [хорошее видео](https://youtu.be/yZYOX6hRYFc), рассказывающее о возможностях этого инструмента

kirillNay commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -0,0 +1,83 @@
+        android:singleLine="true"
+        style="@style/primary_text"/>
+
+    <LinearLayout
```
⏫ Так как мы используем `ConstraintLayout` в качестве самой рутовой `ViewGroup`, мы можем организовать разметку любой сложности без необходимости добавлять вложенность. 

Главное помнить, что любая вложенность элементов будет негативно сказываться на времени отрисовки экрана, поэтому при любой возможности ее лучше избегать

kirillNay commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -0,0 +1,83 @@
+            android:textColor="?attr/color"
+            style="@style/primary_text"/>
+
+        <ImageView
```
⏫ Вместо `ImageView` можно было бы воспользоваться обычным знаком "•")

kirillNay commented on app/src/main/res/layout/track_item.xml:
⏫ Хорошим тоном является добавление тестовых данных для превью в `tools:..` атрибуты

NickFilimonov committed:
pr. Playlist Maker commit 12
Домашняя работа №-10. Доработкию
- fun getItemCount переопредела по стандартам Kotlin
- Поправил отступы функций glide в TrackViewHolder.kt
